### PR TITLE
Add instagram oauth aa1 to master

### PR DESCRIPTION
**Pull request by pinterid**
**Checklist:**

 

-   [x] Have you added an explanation of what your changes do and why you'd like them to be included?
-   [x] Have you updated or added documentation for the change?
-   [x] Have you tested your changes with successful results?

**Type of Changes:**

 

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Documentation (non-breaking change which adds documentation)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

-  Currently it is not possible to authenticate with Instagram.

**What is the new behavior (if this is a feature change)?**

-  Now it is possible to authenticate with Instagram.

**Other information**:

-  Ref: https://developers.facebook.com/docs/instagram-basic-display-api/guides/getting-access-tokens-and-permissions

### CODE REVIEW

File: src/Instagram/index.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  constructor(props) {
+    super(props);
+
+    this.onBtnClick = this.onBtnClick.bind(this);
+    this.onRequest = this.onRequest.bind(this);
+    this.onSuccess = this.onSuccess.bind(this);
+    this.onFailure = this.onFailure.bind(this);
+  }
+
+  onBtnClick() {
```
Comment (Aichnerc) [inline src/Instagram/index.jsx]: When using `this.` I'd use arrow functions.

```suggestion
  onBtnClick = () => {
```

**Review by Aichnerc — APPROVED** [review]
LGTM, but you might want to check what I wrote.

File: src/Instagram/index.jsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    formData.append("grant_type", "authorization_code");
+
+    /* POST request to get the access token from Instagram */
+    await fetch("https://api.instagram.com/oauth/access_token", {
+      method: "POST",
+      body: formData,
+    })
+      .then(async (res) => await res.json())
+      .then(async (res) => {
+        let accessToken = res.access_token;
```
Comment (kleberbaum) [inline src/Instagram/index.jsx:83]: Please add a newline.

**Review by kleberbaum — APPROVED** [review]
Should be ok to merge.

**Pull request merged**